packages/core/strapi/lib/commands/transfer/export.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const {
+  createLocalFileDestinationProvider,
+  createLocalStrapiSourceProvider,
+  createTransferEngine,
+} = require('@strapi/data-transfer');
+const strapi = require('../../Strapi');
+
+const getDefaultExportBackupName = () => `strapi-backup`;
+
+module.exports = async (args) => {
+  if (!args.output) {
+    Object.assign(args, { output: getDefaultExportBackupName() });
+  }
+
+  // From strapi
+  const source = createLocalStrapiSourceProvider({
+    getStrapi() {
+      return strapi().load();
+    },
+  });
+
+  // To file
+  const destination = createLocalFileDestinationProvider({
+    backupFilePath: args.output,
+  });
+
+  // create transfer engine
+  const engine = createTransferEngine(source, destination, {
+    strategy: 'restore',
+    versionMatching: 'minor',
+  });
+
+  try {
+    const result = await engine.transfer();
+    if (!result?.path) throw new Error('Export file not created');
```

Comment:
Not sure this should be the engine's responsibility to return information related to the transfer.
For me, it should be handled by the destination provider or through the listeners (not yet implemented) 
What do you think?

Reply:
Are you thinking something like:
`
destination.on("complete", (result) => {
  console.log("File exported at", result.path);
}

try {
  await engine.transfer(); // triggers destination complete event 
}
catch(e) {
 process.exit(1)
}
`

We will have listeners for sure, it's necessary for progress indicators. And "complete(finish/done)" will be an event too.

But I also think it's reasonable to expect `engine.transfer()` to provide data on the transfer process that just ran. even if it's only passing through data from the provider(s). I also fear that if we only rely on listeners, we could end up with some really confusing code in cases where we need to attempt the transfer again or run multiple transfers -- having destination.on("complete") running engine.transfer() feels wrong (and adding some potential callback hell), whereas running an engine.transfer(), getting a result, and then acting on that result is a lot cleaner.

I'll think about this for a bit and play around with some different scenarios for how the engine could be used.

Reply:
Having the engine return some generic information about the transfer makes sense. I'm just not sure about provider-specific information, except if we put that in a "sub-object"-like property.